### PR TITLE
Disable deletion protection for Autopilot test clusters

### PR DESCRIPTION
**Pull request by gongmax**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug

/kind cleanup

> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
Seems deletion protection is enabled by default for AP clusters, and we saw cluster couldn't be deleted when we updated k8s version for test clusters. This PR disable it explicitly. 
**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by roberthbailey** [comment]
This is changing the value is the terraform module, which changes it for both the test clusters but also for anyone that is using our example module: https://github.com/googleforgames/agones/blob/main/examples/terraform-submodules/gke-autopilot/module.tf

I think it would make more sense to make it a variable, set it to true by default (to keep the current behavior) and set it to false explicitly in the test configs.

**Comment by Kalaiselvi84** [comment]
Are we good to merge this PR?

**Review by zmerlynn — APPROVED** [review]

**Review by zmerlynn — CHANGES_REQUESTED** [review]
Wait. One thing I realized for both this and maintenance exclusions is that we injected it into the `install/terraform` path. I don't think that's right - I think maintenance exclusions should be optional in the base install and we need this in the e2e specific one. Unless I'm mis-remembering the structure here?

**Comment by gongmax** [comment]
> Wait. One thing I realized for both this and maintenance exclusions is that we injected it into the `install/terraform` path. I don't think that's right - I think maintenance exclusions should be optional in the base install and we need this in the e2e specific one. Unless I'm mis-remembering the structure here?

Make sense. Updated accordingly.

File: install/terraform/modules/gke/variables.tf [hunk 1]
```diff
@@ -20,21 +20,23 @@ variable "cluster" {
...
+    "releaseChannel"                = "UNSPECIFIED"
+    "kubernetesVersion"             = "1.27"
+    "windowsInitialNodeCount"       = "0"
+    "windowsMachineType"            = "e2-standard-4"
+    "autoscale"                     = false
+    "workloadIdentity"              = false
+    "minNodeCount"                  = "1"
+    "maxNodeCount"                  = "5"
+    "maintenanceExclusionStartTime" = null
+    "maintenanceExclusionEndTime"   = null
```
Comment (zmerlynn) [inline install/terraform/modules/gke/variables.tf:39]: did you mean to add `deletionProtection` in here as well?

Comment (gongmax) [reply install/terraform/modules/gke/variables.tf:39]: We didn't see any issue with standard cluster last time when we recreate test clusters. So I'd prefer not add it here unless we see the issue next time.

**Review by zmerlynn — APPROVED** [review]

**Pull request merged**